cortex/user_preferences.py

```
@@ -0,0 +1,371 @@
+        }
+        
+        # Write atomically (write to temp, then rename)
+        temp_path = self.config_path.with_suffix('.yaml.tmp')
+        try:
+            with open(temp_path, 'w') as f:
+                yaml.dump(data, f, default_flow_style=False, sort_keys=False)
+            
+            # Atomic rename
+            temp_path.replace(self.config_path)
+            
+        except Exception as e:
```

Ruff (0.14.5)

Do not catch blind exception: `Exception`

(BLE001)